quic/s2n-quic-core/src/havoc.rs

```diff
@@ -354,6 +391,46 @@ impl Strategy for Disabled {
     fn havoc<R: Random>(&mut self, _rand: &mut R, _buffer: &mut EncoderBuffer) {}
 }
 
+#[derive(Clone, Debug, Default)]
+pub struct Hold<S: Strategy> {
+    strategy: S,
+    min: u64,
+    max: u64,
+    value: Vec<u8>,
+    remaining: u64,
+}
+
+impl<S: Strategy> Hold<S> {
+    pub fn new(strategy: S, range: Range<usize>) -> Self {
+        debug_assert!(range.start <= range.end);
+        Self {
+            strategy,
+            min: range.start as _,
+            max: range.end as _,
+            value: Vec::new(),
+            remaining: 0,
+        }
+    }
+}
+
+impl<S: Strategy> Strategy for Hold<S> {
+    #[inline]
+    fn havoc<R: Random>(&mut self, rand: &mut R, buffer: &mut EncoderBuffer) {
+        if self.remaining == 0 {
+            self.strategy.havoc(rand, buffer);
+            // store the value after the strategy has been applied
+            self.value.clear();
+            self.value.extend_from_slice(buffer.as_mut_slice());
+            self.remaining = rand.gen_range(self.min..self.max);
+        } else {
+            // restore the value from the first application of the strategy
+            buffer.set_position(0);
+            buffer.write_slice(self.value.as_mut_slice());
```

Comment:
Probably a good idea to bound the slice that we're writing to the `buffer.capacity()`, just in case that changes between calls.